src/Compiler/CodeGen/IlxGen.fs

```diff
@@ -1859,29 +1859,29 @@ let MergePropertyDefs m ilPropertyDefs =
 
 /// Information collected imperatively for each type definition
 type TypeDefBuilder(tdef: ILTypeDef, tdefDiscards) =
-    let gmethods = ResizeArray<ILMethodDef>(0)
-    let gfields = ResizeArray<ILFieldDef>(0)
+    let gmethods = ResizeArray<ILMethodDef>(tdef.Methods.AsList())
```

Comment:
I guess the idea before was that we don't grow the list with methods from typedef, and only make those in `Close`. I presume it's still *should* be fine, but i'd double check on big types.